archiver_test.go

```diff
@@ -39,6 +39,10 @@ func symmetricTest(t *testing.T, name string, ar Archiver) {
 		t.Fatalf("making archive: didn't expect an error, but got: %v", err)
 	}
 
+	if !ar.Match(outfile) {
+		t.Fatalf("identifying format should be 'true', but got 'false'")
+	}
```

Comment:
At the first commit 183d279 of the patch series, I confirmed the test failed because the `outfile` value is e.g. `test-TarGz`, without an extension at testing .tar.gz. After adding checking file header implementations, the test is passed.